blazor/diagram/data-binding.md

```diff
@@ -654,105 +654,364 @@ To bind remote data to [Diagram component](https://help.syncfusion.com/cr/blazor
-        {
-            node.Style.Fill = "#3A4857";
-        }
-        else if (data["FirstName"].ToString() == "Nancy")
-        {
-            node.Style.Fill = "#2B8C68";
-        }
-        else if (data["FirstName"].ToString() == "Janet")
-        {
-            node.Style.Fill = "#488CC1";
-        }
-        else if (data["FirstName"].ToString() == "Janet")
-        {
-            node.Style.Fill = "#488CC1";
-        }
-        else if (data["FirstName"].ToString() == "Margaret")
-        {
-            node.Style.Fill = "#4C888F";
-        }
-        else if (data["FirstName"].ToString() == "Steven")
-        {
-            node.Style.Fill = "#8E4DB4";
-        }
-        else if (data["FirstName"].ToString() == "Laura")
-        {
-            node.Style.Fill = "#CD6A32";
+
+        switch (data["FirstName"].ToString()) {
+            case "Andrew":
+                node.Style.Fill = "#3A4857";
+                break;
+            case "Nancy":
+                node.Style.Fill = "#2B8C68";
+                break;
+            case "Janet":
+                node.Style.Fill = "#488CC1";
+                break;
+            case "Margaret":
+                node.Style.Fill = "#4C888F";
+                break;
+            case "Steven":
+                node.Style.Fill = "#8E4DB4";
+                break;
+            case "Laura":
+                node.Style.Fill = "#CD6A32";
+                break;
+            default:
+                node.Style.Fill = "#8E4DB4";
+                break;
         }
-        else
-        {
-            node.Style.Fill = "#8E4DB4";
+    }
+
+    private void OnConnectorCreating(IDiagramObject obj) {
+        Connector connector = obj as Connector;
+        connector.Style.StrokeColor = "#048785";
+        connector.Type = ConnectorSegmentType.Orthogonal;
+        connector.TargetDecorator.Shape = DecoratorShape.None;
+        connector.SourceDecorator.Shape = DecoratorShape.None;
+        connector.Style = new ShapeStyle() { StrokeColor = "#3A4857", Fill = "#3A4857", StrokeWidth = 1, StrokeDashArray = "3,3" };
+    }
+
+}
+```
+You can download a complete working sample from [GitHub](https://github.com/SyncfusionExamples/Blazor-Diagram-Examples/tree/master/UG-Samples/DataBinding/RemoteData)
+
+### Binding with WebAPI services
+
+You can use the `WebApiAdaptor` to interact with Web APIs created with OData endpoint. The `WebApiAdaptor` is extended from the `ODataAdaptor`. Hence to use `WebApiAdaptor`, the endpoint should understand the OData formatted queries sent along with request.
+
+To enable OData query option for Web API, refer to this [documentation](https://learn.microsoft.com/en-us/aspnet/web-api/overview/odata-support-in-aspnet-web-api/supporting-odata-query-options).
+
+The following sample code demonstrates binding remote data to the Diagram component through the [SfDataManager](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.Data.SfDataManager.html) using Web API service,
+
```

Comment:
You can use the WebApiAdaptor to interact with Web APIs created with OData endpoints. The WebApiAdaptor is extended from the ODataAdaptor. Therefore, to use the WebApiAdaptor, the endpoint must understand the OData formatted queries sent with the request.

To enable OData query options for a Web API, refer to this [documentation](https://learn.microsoft.com/en-us/aspnet/web-api/overview/odata-support-in-aspnet-web-api/supporting-odata-query-options).

The following sample code demonstrates how to bind remote data to the Diagram component through the [SfDataManager](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.Data.SfDataManager.html) using a Web API service: